### PR TITLE
Fix issue #4136: 'Restructuring documentation'

### LINKED ISSUE #4136: Restructuring documentation
**What problem or use case are you trying to solve?**

It would be best to restructure the OpenHands documentation to make it a bit easier to parse.
This can mostly be done by reordering `sidebars.ts` for now, without changing content.

**Proposal**

Here is a proposal of top-level and second-level menus:

* Getting Started
* Usage Methods
  * CLI Mode
  * Headless Mode
  * Github Actions
* Advanced Configuration
  * LLM Configuration (previously "LLMs")
  * Custom Sandbox
* Troubleshooting
* Feedback
* For OpenHands Developers
  * Architecture
  * Debugging
  * Evaluation
  * Kubernetes Deployment (previously "Kubernetes")
* About

### PR DESCRIPTION
**Pull request by neubig**
- [x] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below

**End-user friendly description of the problem this fixes or functionality that this introduces**
Restructure documentation to be in an easier-to-follow format

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**
Mostly self-explanatory, see #4136 for details

---
**Link of any specific issues this addresses**
Fixes #4136

### CODE REVIEW

File: docs/sidebars.ts [hunk 1]
```diff
@@ -2,25 +2,46 @@ import type { SidebarsConfig } from "@docusaurus/plugin-content-docs";
 
 const sidebars: SidebarsConfig = {
   apiSidebar: [require("./modules/python/sidebar.json")],
-  docsSidebar: [
+docsSidebar: [
```
Comment (tobitege) [inline docs/sidebars.ts:5]: indentation seems off

Comment (neubig) [reply docs/sidebars.ts]: Thanks, fixed.

**Comment by mamoodi** [comment]
Couple of notes:
* `Usage Methods` shouldn't be that high because most people will not be using CLI, Headless or Github Actions yet.
* `LLM Configuration` doesn't really fall under `Advanced configuration`. The LLM Configuration is for the most basic configuration of OpenHands with the most common providers. This should really still be right under `Getting Started` as that is essential for getting OpenHands setup. The `Custom Sandbox` makes sense as an `Advanced Configuration`

Rest looks good.

**Review by tofarr — APPROVED** [review]
🍰

**Comment by neubig** [comment]
PS: I just realized that I should open this issue about creating a separate page for "GUI" on top of the "Usage Methods" page, which is something I'm thinking of doing after this initial restructure: https://github.com/All-Hands-AI/OpenHands/issues/4142

**Comment by mamoodi** [comment]
> Hey @mamoodi , thanks!
> 
> 1. I'd like to hear other's (particularly @rbren's) take on this as well, but I think we'd kinda like to meet developers where they're at, give them different options, and see which one people like the most. That was the thinking about this.
> 2. For basic LLM configuration, this is already included in "Getting Started". I feel that we're trying to move away from getting people to think about lots of different LMs, because only Claude works well at the moment (and maybe gpt-4o).
> 
> What do you think?

So is our stance that OpenHands only works with Claude Sonnet 3.5 at the moment?

File: docs/sidebars.ts [hunk 2]
```diff
@@ -62,48 +83,28 @@ const sidebars: SidebarsConfig = {
...
-      items: [
-        {
-          type: 'doc',
-          label: 'Backend',
-          id: 'usage/architecture/backend',
-        },
-        {
-          type: 'doc',
-          label: 'Runtime',
-          id: 'usage/architecture/runtime',
```
Comment (tobitege) [inline docs/sidebars.ts:105]: Isn't the runtime an important part to keep included?

File: docs/sidebars.ts [hunk 3]
```diff
@@ -10,17 +10,33 @@ const sidebars: SidebarsConfig = {
...
         },
+      ],
+    },
+    {
+      type: 'category',
+      label: 'Advanced Configuration',
+      items: [
         {
           type: 'category',
-          label: 'Providers',
```
[the remark this answers is not in the document]
Comment (neubig) [reply docs/sidebars.ts:22]: Oh no, sorry, that was a mistake! Fixed.

**Comment by neubig** [comment]
> So is our stance that OpenHands only works with Claude Sonnet 3.5 at the moment?

This is oversimplifying a bit (and we're working on publishing some numbers on which LMs work and which don't), but until we're able to make concrete recommendations about this I think that defaulting to only recommending Claude Sonnet 3.5 seems like a good idea.

**Comment by mamoodi** [comment]
> > So is our stance that OpenHands only works with Claude Sonnet 3.5 at the moment?
> 
> This is oversimplifying a bit (and we're working on publishing some numbers on which LMs work and which don't), but until we're able to make concrete recommendations about this I think that defaulting to only recommending Claude Sonnet 3.5 seems like a good idea.

Alright. Cool. Just want to know what to tell people.

**Review by xingyaoww — APPROVED** [review]

**Comment by rbren** [comment]
I agree w/ Graham--while most people are using the UI, we'd like to start highlighting other ways of interacting with OH. So I think it's worth moving those up.

Re: LLMs, I do think we need better "LLM setup" docs for a few major providers (at least Anthropic and OpenAI--i.e. where do you get a key, etc). Maybe we can expand the getting started docs with specific OpenAI and Anthropic links, and then move the other LLMs into "Advanced"?

**Comment by neubig** [comment]
Opened an issue for the second one: https://github.com/All-Hands-AI/OpenHands/issues/4146

**Pull request merged**